Sofa/Component/Constraint/Lagrangian/Solver/src/sofa/component/constraint/lagrangian/solver/ConstraintForceExporter.inl

```diff
@@ -0,0 +1,148 @@
+{
+    Inherited::init();
+
+    if (!l_constraintSolver)
+    {
+        l_constraintSolver.set(this->getContext()->template get<ConstraintSolverImpl>());
+    }
+
+    if (!l_constraintSolver)
+    {
+        msg_error() << "No constraint solver found in the current context, whereas it is required. This component retrieves the matrix from a linear solver.";
+        this->d_componentState.setValue(sofa::core::objectmodel::ComponentState::Invalid);
+        return;
+    }
+
+    // No need to check if the link to the mechanical state is correct
+    // as it has been covered by canCreate() 
+
+    this->d_componentState.setValue(sofa::core::objectmodel::ComponentState::Valid);
+}
+
+template <class DataTypes>
+void ConstraintForceExporter<DataTypes>::handleEvent(core::objectmodel::Event* event)
+{
+    if (this->d_componentState.getValue() != sofa::core::objectmodel::ComponentState::Valid)
+    {
+        return;
+    }
+
+    Inherited::handleEvent(event);
+
+    if (simulation::AnimateEndEvent::checkEventType(event))
+    {
+        computeForce();
+    }
+}
+
+template <class DataTypes>
+void ConstraintForceExporter<DataTypes>::draw(const sofa::core::visual::VisualParams* vparams)
+{
+    if (!d_draw.getValue() || this->d_componentState.getValue() != sofa::core::objectmodel::ComponentState::Valid)
+    {
+        return;
+    }
+
+    auto constraintForces = sofa::helper::getReadAccessor(d_constraintForces);
+    const auto& positions = l_mechanicalState->read(sofa::core::ConstVecCoordId::position())->getValue();
+
+    assert(constraintForces.size() != positions.size());
+
+    const float lineSize = 2.0f;
+    const double forceScale = d_drawForceScale.getValue();
+
+    vparams->drawTool()->saveLastState();
+
+    std::vector<type::Vector3> vertices;
+    vertices.reserve(constraintForces.size() * 2);
+
+    for(std::size_t i = 0 ; i < constraintForces.size() ; i++)
+    {
+        const auto& origin = positions[i];
+        const auto& force = DataTypes::getDPos(constraintForces[i]);
+
+        vertices.emplace_back( origin[0],
+                               origin[1],
+                               origin[2] );
+        vertices.emplace_back( origin[0] + force[0] * forceScale,
+                               origin[1] + force[1] * forceScale,
+                               origin[2] + force[2] * forceScale );
+    }
+    vparams->drawTool()->drawLines(vertices, lineSize, sofa::type::RGBAColor::white());
+
+    vparams->drawTool()->restoreLastState();
+}
+
+template <class DataTypes>
+void ConstraintForceExporter<DataTypes>::computeForce()
+{
+    auto constraintForces = sofa::helper::getWriteOnlyAccessor(d_constraintForces);
+
+    auto* constraintProblem = l_constraintSolver->getConstraintProblem();
+    const auto& constraintMatrix = l_mechanicalState->read(core::ConstMatrixDerivId::constraintJacobian())->getValue();
+    auto* lambdas = constraintProblem->getF();
+    int dimension = constraintProblem->getDimension();
```

Comment:
```suggestion
    [[maybe_unused]] const int dimension = constraintProblem->getDimension();
```

Note that `dimension` will be used only in debug mode. So it could trigger a warning in the other compilation modes saying that the variable is unused. You can add a `maybe_unused` attribute to avoid this warning.